rc/trafficserver.service.in

```diff
@@ -22,9 +22,16 @@ After=syslog.target network.target
 [Service]
 Type=simple
 EnvironmentFile=-/etc/sysconfig/trafficserver
-PIDFile=@exp_runtimedir@/manager.pid
-ExecStart=@exp_bindir@/traffic_manager --bind_stdout @exp_logdir@/traffic.out --bind_stderr @exp_logdir@/traffic.out $TM_DAEMON_ARGS
+ExecStart=@exp_bindir@/traffic_manager $TM_DAEMON_ARGS
+Restart=on-failure
+RestartSec=5s
+ExecStopPost=/bin/sh -c ' \
+        export TM_PIDFILE=$(@exp_bindir@/traffic_layout} 2>/dev/null | grep RUNTIMEDIR | cut -d: -f2)/manager.lock ; \
```

Comment:
How stable is this across differing operating systems?

Reply:
We are calling traffic_layout to get the data... I hope we are stable :-)